packages/azure-services/src/azure-cosmos/cosmos-client-wrapper.ts

```diff
@@ -4,26 +4,49 @@ import * as cosmos from '@azure/cosmos';
 import { System } from 'common';
 import { inject, injectable } from 'inversify';
 import * as _ from 'lodash';
+import { Logger } from 'logger';
 import { CosmosClientProvider, iocTypeNames } from '../ioc-types';
+import { client } from '../storage/client';
 import { CosmosDocument } from './cosmos-document';
 import { CosmosOperationResponse } from './cosmos-operation-response';
 
+// tslint:disable: no-any no-unsafe-any
+
+export declare type CosmosOperation = 'upsertItem' | 'readAllItems' | 'queryItems' | 'readItem' | 'deleteItem';
+
 @injectable()
 export class CosmosClientWrapper {
     public static readonly PARTITION_KEY_NAME: string = '/partitionKey';
     public static readonly MAXIMUM_ITEM_COUNT: number = 100;
 
-    constructor(@inject(iocTypeNames.CosmosClientProvider) private readonly cosmosClientProvider: CosmosClientProvider) {}
+    constructor(
+        @inject(iocTypeNames.CosmosClientProvider) private readonly cosmosClientProvider: CosmosClientProvider,
+        @inject(Logger) private readonly logger: Logger,
+    ) {}
 
-    public async upsertItems<T>(items: T[], dbName: string, collectionName: string, partitionKey?: string): Promise<void> {
+    public async upsertItems<T extends CosmosDocument>(
+        items: T[],
+        dbName: string,
+        collectionName: string,
+        partitionKey?: string,
+    ): Promise<void> {
         const container = await this.getContainer(dbName, collectionName);
-
         const chunks = System.chunkArray(items, 10);
 
         for (const chunk of chunks) {
             await Promise.all(
                 chunk.map(async item => {
-                    await container.items.upsert(item, this.getOptions(item, partitionKey));
+                    try {
```

Comment:
can you make this class to take in context aware logger so that we tag the logs with the properties that match the request?

Reply:
for calls from batch, this argument will be same as main logger. for web api calls, we should pass the contextAwareLogger